core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java

```diff
@@ -70,4 +74,15 @@ public void doFilterInternal(@SuppressWarnings("NullableProblems") HttpServletRe
         }
         chain.doFilter(request, response);
     }
+
+    /**
+     * Convert comma separated string of {@link User} roles to collection of roles.
+     *
+     * @param user - {@link Optional} of {@link User}
+     * @return {@link Collection} of roles
+     */
+    private Collection<? extends GrantedAuthority> convertRoles(Optional<User> user) {
+        return user.map(value -> AuthorityUtils.commaSeparatedStringToAuthorityList(value.getRole().name()))
```

Comment:
Yes, this is what I meant. Taking into account that now it is one line and that passing an Optional is considered a bad practice, you can just inline this method.